src/main/java/com/gluonhq/richtext/RichTextAreaSkin.java

```diff
@@ -291,9 +295,13 @@ private void addBackgroundPathsToLayers(List<IndexRangeColor> backgroundIndexRan
                     path.setLayoutY(textFlowLayoutY);
                     return path;
                 })
-                .collect(Collectors.toList());
-        textBackgroundColorPaths.removeIf(path -> !backgroundPaths.contains(path));
-        textBackgroundColorPaths.addAll(backgroundPaths);
+                .forEach(path -> fillPathMap.merge(path.getFill(), path, (p1, p2) -> {
+                    Path union = (Path) Shape.union(p1, p2);
+                    union.setFill(p1.getFill());
+                    return union;
+                }));
+        textBackgroundColorPaths.removeIf(path -> !fillPathMap.containsValue(path));
+        textBackgroundColorPaths.addAll(fillPathMap.values());
```

Comment:
Could be simplified to one loop:

```
for (IndexRangeColor indexRangeBackground : backgroundIndexRanges) {
    final Path path = new BackgroundColorPath(textFlow.rangeShape(indexRangeBackground.getStart(), indexRangeBackground.getEnd()));
    ...
    fillPathMap.merge(path.getFill(), path, (p1, p2) -> {
        Path union = (Path) Shape.union(p1, p2);
        union.setFill(p1.getFill());
        return union;
    });
}
```

Reply:
It is already one stream call, so I don't see  a difference?

Reply:
Sorry, but I still don't see a real difference between:

```
Map<Paint, Path> fillPathMap = new HashMap<>();
for (IndexRangeColor indexRangeBackground : backgroundIndexRanges) {
    final Path path = new BackgroundColorPath(textFlow.rangeShape(indexRangeBackground.getStart(), indexRangeBackground.getEnd()));
    ...
    fillPathMap.merge(path.getFill(), path, (p1, p2) -> {
        Path union = (Path) Shape.union(p1, p2);
        union.setFill(p1.getFill());
        return union;
    });
}
```
and
```
Map<Paint, Path> fillPathMap = new HashMap<>();
backgroundIndexRanges.stream()
                 .map(indexRangeBackground -> {
                      final Path path = new BackgroundColorPath(textFlow.rangeShape(indexRangeBackground.getStart(), indexRangeBackground.getEnd()));
    ...
                      return path;
                 })
               .forEach(path -> fillPathMap.merge(path.getFill(), path, (p1, p2) -> {
                     Path union = (Path) Shape.union(p1, p2);
                     union.setFill(p1.getFill());
                     return union;
                 }));
```

What am I missing?